src/Explorer/Notebook/NotebookRenderer/outputs/IFrameOutputs.tsx

```diff
@@ -0,0 +1,103 @@
+import { AppState, ContentRef, selectors } from "@nteract/core";
+import { Output } from "@nteract/outputs";
+import Immutable from "immutable";
+import React from "react";
+import ReactDOM from "react-dom";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import * as _ from "underscore";
+import { copyStyles } from "../../../../Utils/StyleUtils";
+
+// Adapted from https://github.com/nteract/nteract/blob/main/packages/stateful-components/src/outputs/index.tsx
+// to add support for sandboxing using <iframe>
+
+interface ComponentProps {
+  id: string;
+  contentRef: ContentRef;
+  children: React.ReactNode;
+}
+
+interface StateProps {
+  hidden: boolean;
+  expanded: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  outputs: Immutable.List<any>;
+}
+
+const StyledIFrame = styled.iframe`
+  width: 100%;
+  border-style: unset;
+  sandbox: allow-downloads allow-forms allow-pointer-lock allow-popups allow-scripts allow-popups-to-escape-sandbox;
+`;
+
+export class IFrameOutputs extends React.PureComponent<ComponentProps & StateProps> {
+  frame: HTMLIFrameElement;
+  outputsContainer: HTMLElement;
+  resizeObserver: ResizeObserver;
+
+  render(): JSX.Element {
+    return (
+      <StyledIFrame ref={(frame) => (this.frame = frame)}>
+        {this.outputsContainer && ReactDOM.createPortal(this.renderOutputs(), this.outputsContainer)}
+      </StyledIFrame>
+    );
+  }
+
+  componentDidMount(): void {
+    copyStyles(document, this.frame.contentWindow.document);
+
+    const { hidden, expanded } = this.props;
+    this.outputsContainer = document.createElement("div");
+    this.outputsContainer.className = `nteract-cell-outputs ${hidden ? "hidden" : ""} ${expanded ? "expanded" : ""}`;
+
+    this.resizeObserver = new ResizeObserver(
+      _.debounce(() => (this.frame.height = this.outputsContainer.scrollHeight + "px"), 200)
+    );
+    this.resizeObserver.observe(this.outputsContainer);
```

Comment:
What removes this observer? Will this continue to observe after the component unmounts?

Reply:
Added code to disconnect this observer when component unmounts.